File: NGOISAO6.py
'''
Viết chương trình in ra màn hình tam giác sao rỗng (nghĩa là chỉ in đường viền cạnh của tam giác) với chiều cao h nhập vào từ bàn phím.

Ví dụ:
Input:
3

Output:
  *  
 * * 
*****

Input:
5

Output:
    *    
   * *   
  *   *  
 *     * 
*********

Input:
6

Output:
     *     
    * *    
   *   *   
  *     *  
 *       * 
***********
Lưu ý: 

Xuất chính xác từng ký tự. Không xuất dư hoặc thiếu khoảng trắng, không xuất dư endl dòng cuối cùng.
'''

h = int(input())
for i in range(h):
    line = ""
    for j in range(2 * h - 1):
        if i == h - 1:
            line += "*"
        else:
            if j == h - i - 1 or j == h + i - 1:
                line += "*"
            else:
                line += " "
    print(line, end="" if i == h - 1 else "\n")

File: LTCB/C++/VongLap/Revise/NGOISAO6.cpp
/*
Viết chương trình in ra màn hình tam giác sao rỗng (nghĩa là chỉ in đường viền cạnh của tam giác) với chiều cao h nhập vào từ bàn phím.

Ví dụ:
Input:
3

Output:
  *  
 * * 
*****

Input:
5

Output:
    *    
   * *   
  *   *  
 *     * 
*********

Input:
6

Output:
     *     
    * *    
   *   *   
  *     *  
 *       * 
***********
Lưu ý: 

Xuất chính xác từng ký tự. Không xuất dư hoặc thiếu khoảng trắng, không xuất dư endl dòng cuối cùng.
*/

#include <iostream>
using namespace std;

int main(){
    int h;
    cin >> h;
    for(int i = 1; i <= h; i++){
        // In khoảng trắng trước ngôi sao
        for(int j = 1; j <= h - i; j++){
            cout << " ";
        }
        // In ngôi sao
        for(int j = 1; j <= 2*i - 1; j++){
            if(j == 1 || j == 2*i -1 || i == h){ // In ngôi sao ở đầu và cuối mỗi hàng, và ở hàng cuối cùng
                cout << "*";
            }else {
                cout << " ";
            }
        }
        // In khoảng trắng sau ngôi sao
        for(int j = 1; j <= h - i; j++){
            cout << " ";
        }
        if(i < h) cout << endl;
    }
    return 0;
}